cloud_function/PeMS/PeMS.py

```diff
@@ -0,0 +1,84 @@
+import requests
+from bs4 import BeautifulSoup
+
+BASE_URL = "https://pems.dot.ca.gov"
+USERNAME = ""
+PASSWORD = ""
```

Comment:
Not recommended for "real world" code! This can be hard to get right, but nowdays cloud providers like GCP provide online secret managers that can be accessed through either personal credentials (for developers working off a personal computer) or service accounts (for online instances / services).

Reply:
It's done. All credentials are passed through environment variables.